sdk/python/kfp/compiler/compiler.py

```diff
@@ -542,12 +548,61 @@ def _compile(self, pipeline_func):
     args_list_with_defaults = [dsl.PipelineParam(self._sanitize_name(arg_name))
                                for arg_name in argspec.args]
     if argspec.defaults:
-      for arg, default in zip(reversed(args_list_with_defaults), reversed(argspec.defaults)):
+      for arg, default in zip(reversed(args_list_with_defaults),
+                              reversed(argspec.defaults)):
         arg.value = default.value
 
     workflow = self._create_pipeline_workflow(args_list_with_defaults, p)
     return workflow
 
+  def _convert_k8s_obj_to_dic(self, obj):
+    """
+    Builds a JSON K8s object.
+
+    If obj is None, return None.
+    If obj is str, int, long, float, bool, return directly.
+    If obj is datetime.datetime, datetime.date
+        convert to string in iso8601 format.
+    If obj is list, sanitize each element in the list.
+    If obj is dict, return the dict.
+    If obj is swagger model, return the properties dict.
+
+    :param obj: The data to serialize.
+    :return: The serialized form of data.
+    """
+
+    from six import text_type, integer_types
+    PRIMITIVE_TYPES = (float, bool, bytes, text_type) + integer_types
+    from datetime import date, datetime
+    if obj is None:
+      return None
+    elif isinstance(obj, PRIMITIVE_TYPES):
+      return obj
+    elif isinstance(obj, list):
+      return [self._convert_k8s_obj_to_dic(sub_obj)
+              for sub_obj in obj]
+    elif isinstance(obj, tuple):
+      return tuple(self._convert_k8s_obj_to_dic(sub_obj)
+                   for sub_obj in obj)
+    elif isinstance(obj, (datetime, date)):
+      return obj.isoformat()
+
+    if isinstance(obj, dict):
+      obj_dict = obj
+    else:
+      # Convert model obj to dict except
+      # attributes `swagger_types`, `attribute_map`
+      # and attributes which value is not None.
+      # Convert attribute name to json key in
+      # model definition for request.
+      from six import iteritems
+      obj_dict = {obj.attribute_map[attr]: getattr(obj, attr)
+                  for attr, _ in iteritems(obj.swagger_types)
+                  if getattr(obj, attr) is not None}
+
+    return {key: self._convert_k8s_obj_to_dic(val)
```

Comment:
it was a direct copy from 
https://github.com/kubernetes-client/python/blob/master/kubernetes/client/api_client.py#L172
to convert the snake case k8s object key to camel case

Reply:
This function is directly taken from K8s' ApiClient.
The key should be correct, because it was taken from `obj.attribute_map[attr]` which has the correct key names. (See Line 599).

Note: As the name says, it's intended to convert k8s **objects**. It only needs to correct the keys when converting an object (and not, say, dict)